File: task3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        char_index = {}  # символ -> последняя позиция + 1
        max_len = 0
        left = 0
        
        for right, c in enumerate(s):
            # Если символ уже в окне — сдвигаем левую границу
            if c in char_index and char_index[c] > left:
                left = char_index[c]
            
            char_index[c] = right + 1
            max_len = max(max_len, right - left + 1)
        
        return max_len

File: task3.c
int lengthOfLongestSubstring(char* s) {
    int charIndex[128] = {0};
    
    int maxLen = 0;
    int left = 0;
    
    for (int right = 0; s[right] != '\0'; right++) {
        char c = s[right];

        if (charIndex[(int)c] > left) {
            left = charIndex[(int)c];
        }

        charIndex[(int)c] = right + 1;
        
        int currentLen = right - left + 1;
        if (currentLen > maxLen) {
            maxLen = currentLen;
        }
    }
    
    return maxLen;
}
